### DISCUSSION TITLE
Start a github workflow on a tag name and build based on that specific tag

### DISCUSSION THREAD

**Discussion by oth-libs** [community/community#25566] [Actions]
<p>I want to start my workflow on tags that contains the value “buildRelease” so I’m using ‘*buildRelease*’, in practice the branch name would be something like “2.0.1-buildRelease”, “2.0.2-buildRelease”, how do I specify in the workflow config that I want to build with the code base of “2.0.2-buildRelease” and not the latest image of the code.</p>
<pre><code>on:
push:
tags:
-'\*buildRelease\*'
</code></pre>
<p>I’m checking out the project like this: </p>
<pre><code>-uses: actions/checkout@v1
with:
ref:'branch-name'
</code></pre>
<p>So instead of “branch-name” I want to point to the tag name that triggered the workflow.</p>
<p>Any idea?</p>

**Comment by BrightRan** [comment]
<p>Hi <a class="mention" href="/oth-libs">@oth-libs</a> ,</p>
<p>Please used the following syntax to specify the tag ‘2.0.2-buildRelease’ for checkout action in the workflow run.</p>
<pre><code>- uses: actions/checkout@v1
   with:
     ref: refs/tags/2.0.2-buildRelease
</code></pre>

**Comment by oth-libs** [comment, marked as answer]
<p>Thanks <a class="mention" href="/brightran">@brightran</a> , but I just figured out what I needed. it was actually more like this: </p>
<pre><code>on:
  push:
    tags:
      - '*productionDebug*'
..
..

      - name: Get tag name
        id: get_tag_name
        run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
      - uses: actions/checkout@v1
        with:
          ref: ${{ steps.get_tag_name.outputs.VERSION }}
</code></pre>
<p>so I wanted to trigger the build with a tag containing a keyword and then building based on that whole tag name, while keeping it generic.</p>
<p>cheers</p>

**Comment by uhurusurfa** [comment]
You can now get the trigger event reference from the context:
https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

Look for github.ref_name
It would look like this:
`ref: ${{ github.ref_name}}`
